src/autoscaler/metricscollector/collector/apppoller_test.go

```diff
@@ -211,7 +211,7 @@ var _ = Describe("Apppoller", func() {
 					Expect(appid).To(Equal("test-app-id"))
 					Expect(token).To(Equal("bearer test-access-token"))
 
-					if noaa.ContainerEnvelopesCallCount()%2 == 0 {
+					if noaa.ContainerEnvelopesCallCount() >= 2 && noaa.ContainerEnvelopesCallCount() <= 4 {
```

Comment:
The range is required because the idea was to simulate "a poll success and the corresponding three poll failures and thereafter success again". Basically, we pass the first time (when call count is 1) and then fail 3 times (between 2 to 4) and allow a pass again after that.